### ISSUE TITLE
NoLineWrap: no support for module declarations

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21161]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/whitespace/nolinewrap.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="NoLineWrap"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > nlw/module-info.java <<'EOF'
module com.example
        .app {
    requires java.base;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml nlw/module-info.java

Starting audit...
Audit done.
```

The same wrapping on a package or import statement is reported:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > A.java <<'EOF'
package com.example
        .sub;

import java.util
        .List;

class A {
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml A.java

Starting audit...
[ERROR] A.java:1:1: package statement should not be line-wrapped. [NoLineWrap]
[ERROR] A.java:4:1: import statement should not be line-wrapped. [NoLineWrap]
Audit done.
Checkstyle ends with 2 errors.
```

**Describe what you expect in detail.**

A module declaration should not be line-wrapped either, so the module input should be reported.

The Check's acceptable tokens are `IMPORT`, `STATIC_IMPORT`, `MODULE_IMPORT`, `PACKAGE_DEF`, `CLASS_DEF`, `METHOD_DEF` and others, with no `MODULE_DEF`, so a module declaration is never visited. 

Expected on the module input:

```
1:1: module declaration should not be line-wrapped.
```

New for Check: `TokenTypes.MODULE_DEF`

**Comment by mahfouz72** [comment]
Fix is merged at https://github.com/checkstyle/checkstyle/pull/21233
